### PR TITLE
API cleanup pass

### PR DESCRIPTION
**Pull request by gampleman**
So having gone through [the spreadsheet](https://docs.google.com/spreadsheets/d/e/2PACX-1vQSY_O7Ukz-Q8JZjqRGh9EKM8skcngt83kC-Jgd1U32sCvBOH7IpF5pLCZJ_V5XI1rA8jtz-cMGeWS2/pubhtml), I think I have done most of the more obvious breaking API changes that needed to be done. The basic summary looks like this:

- `Array.Extra.apply` becomes `Array.Extra.andMap` with flipped arguments
- `Array.Extra.interweave`, becomes `Array.Extra.interweave_` with flipped arguments (and a plan that one future day we will then rename that function back to `interweave`.)
- `Basics.Extra.swap` becomes `Tuple.Extra.flip`
- `Basics.Extra.atMost` is removed, since `min` is entirely sufficient
- `Basics.Extra.atLeast` is removed, since `max` is entirely sufficient
- `Basics.Extra.fractionalModBy` becomes `Float.Extra.modBy`
- `Basics.Extra.orderBy` becomes `Order.Extra.breakTies`
- `Basics.Extra.toOrder` becomes `Order.Extra.byField`
- `Basics.Extra.toOrderDesc` is gone and `Order.Extra.byField >> Order.Extra.reverse` is recommended as a replacement
- `Dict.Extra.fromListDedupe` becomes `Dict.Extra.fromListCombining`
- `Dict.Extra.fromListDedupeBy` becomes `Dict.Extra.fromListByCombining`
- `List.Extra.filterNot` becomes `List.Extra.removeWhen`
- `Maybe.Extra.traverse` becomes `Maybe.Extra.combineMap`
- `Maybe.Extra.traverseArray` becomes `Maybe.Extra.combineMapArray`
- `Result.Extra.singleton` is removed, since `Ok` is entirely sufficient
- `String.Extra.removeAccents` becomes `String.Extra.removeDiacritics` (**caveat:** `removeDiacritics` is a more powerful implementation, so while it passes all the tests of `removeAccents`, it will do more removals)
- `Tuple.Extra.sequenceMaybe` becomes `Maybe.Extra.combineBoth`
- `Tuple.Extra.sequenceFirstMaybe` becomes `Maybe.Extra.combineFirst`
- `Tuple.Extra.sequenceSecondMaybe` becomes `Maybe.Extra.combineSecond`
    

This leaves both `Maybe` and `Result` with an identical set of predictably named functions (some of which I have added):

`combine`, `combineMap`, `combineArray`, `combineMapArray`, `combineFirst`, `combineSecond`, `combineBoth`, `combineMapFirst`, `combineMapSecond`, `combineMapBoth`

which I think is neat.

`Dict.Extra.all` is added to complement `Dict.Extra.any`.

Finally, I've changed some documentation formatting to be more consistent.

### CODE REVIEW

File: src/Array/Extra.elm [hunk 1]
```diff
@@ -189,12 +195,36 @@ If one `Array` is longer, its extra elements are not used.
...
 
 
+{-| Map functions taking multiple arguments over multiple arrays. Each array should be of the same length; extra elements are dropped.
+
+    import Array exposing (Array)
+
+    toIntFunctions : Array (Float -> Int)
+    toIntFunctions =
+        Array.fromList [ round
+        , floor
```
Comment (lue-bird) [inline src/Array/Extra.elm:213]: formatting is off

File: src/Array/Extra.elm [hunk 2]
```diff
@@ -767,9 +797,34 @@ Extra elements of either `Array` are glued to the end without anything in betwee
         |> interweave (repeat 1 "on")
     --> fromList [ "turtles", "on", "turtles", "turtles" ]
 
+@deprecated **Beware:** For historical reasons, this function takes it's arguments in the opposite order to `List.Extra.interweave`.
+As such, this function is deprecated in v1 and will be removed in v2. You should switch to `interweave_` which has the correct argument order. We plan to re-introduct `interweave` with the new argument order in a future release, but since the chance of subtle breakage is quite high, we will only do this gradually.
```
Comment (lue-bird) [inline src/Array/Extra.elm:801]: re-introduct → re-introduce

File: src/Array/Extra.elm [hunk 3]
```diff
@@ -767,9 +797,34 @@ Extra elements of either `Array` are glued to the end without anything in betwee
...
+
 -}
 interweave : Array a -> Array a -> Array a
 interweave toInterweave array =
+    interweave_ array toInterweave
+
+
+{-| Place all elements of a given `Array` between all current elements.
```
Comment (lue-bird) [inline src/Array/Extra.elm:809]: With this argument order (non-pipe-friendly), the description seems off. Maybe change it to something like
> Place all elements of the second `Array` between all elements of the first.

File: src/Array/Extra.elm [hunk 4]
```diff
@@ -767,9 +797,34 @@ Extra elements of either `Array` are glued to the end without anything in betwee
...
+    interweave_ (fromList [ "turtles", "turtles", "turtles" ]) (repeat 2 "on")
+    --> fromList [ "turtles", "on", "turtles", "on", "turtles" ]
+
+    interweave_ (fromList [ "turtles", "turtles", "turtles" ]) (repeat 5 "on")
+    --> fromList [ "turtles", "on", "turtles", "on", "turtles", "on", "on", "on" ]
+
+    interweave_ (fromList [ "turtles", "turtles", "turtles" ]) (repeat 1 "on")
+    --> fromList [ "turtles", "on", "turtles", "turtles" ]
+
+See documentation for `interweave` to find out why the strange name.
```
Comment (lue-bird) [inline src/Array/Extra.elm:823]: Add the link as a quality of life: `interweave` → [`interweave`](#interweave)

**Review by lue-bird — COMMENTED** [review]
I only reviewed the Array.Extra changes in detail but the other listed changes also seem wonderful.

**Comment by miniBill** [comment]
Is this meant for v1 or v2?

**Comment by gampleman** [comment]
V1 as there is full backwards compatibility.

File: src/Array/Extra.elm [hunk 5]
```diff
@@ -798,16 +799,16 @@ Extra elements of either `Array` are glued to the end without anything in betwee
     --> fromList [ "turtles", "on", "turtles", "turtles" ]
 
 @deprecated **Beware:** For historical reasons, this function takes it's arguments in the opposite order to `List.Extra.interweave`.
-As such, this function is deprecated in v1 and will be removed in v2. You should switch to `interweave_` which has the correct argument order. We plan to re-introduct `interweave` with the new argument order in a future release, but since the chance of subtle breakage is quite high, we will only do this gradually.
+As such, this function is deprecated in v1 and will be removed in v2. You should switch to `interweave_` which has the correct argument order. We plan to re-introduce `interweave` with the new argument order in a future release, but since the chance of subtle breakage is quite high, we will only do this gradually.
```
Comment (lue-bird) [inline src/Array/Extra.elm:802]: adding links to the `interweave_` references would also be cool

**Pull request merged**